core/server/models/post.js

```diff
@@ -451,8 +453,23 @@ Post = ghostBookshelf.Model.extend({
             }
         }
 
-        if (!options.columns || (options.columns && options.columns.indexOf('url') > -1)) {
-            attrs.url = urlService.getUrlByResourceId(attrs.id);
+        if (options.columns && !options.columns.includes('url')) {
+            delete attrs.url;
+        }
+
+        if (options && options.context && options.context.public) {
+            if (attrs.feature_image) {
+                attrs.feature_image = urlFor('image', {image: attrs.feature_image}, true);
```

Comment:
If we return absolute urls for all images on public access, there are some more usages in the code base which can be removed i think?

If i search for `urlFor('image` i get a lot of usages in our meta data module. Have you talked about these usages?
e.g. [see](https://github.com/TryGhost/Ghost/blob/aedb6dbb90793578f4fbe385e4bc145798fb3ca4/core/server/data/meta/og_image.js)